services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/ColumnarRocksDbKeyValueStorage.java

```diff
@@ -0,0 +1,326 @@
+                      .setBackgroundThreads(rocksDbConfiguration.getBackgroundThreadCount()));
+
+      txOptions = new TransactionDBOptions();
+      final List<ColumnFamilyHandle> columnHandles = new ArrayList<>(columnDescriptors.size());
+      db =
+          TransactionDB.open(
+              options,
+              txOptions,
+              rocksDbConfiguration.getDatabaseDir().toString(),
+              columnDescriptors,
+              columnHandles);
+
+      final Map<BytesValue, String> segmentsById =
+          segments.stream()
+              .collect(
+                  Collectors.toMap(segment -> BytesValue.wrap(segment.getId()), Segment::getName));
+
+      final ImmutableMap.Builder<String, ColumnFamilyHandle> builder = ImmutableMap.builder();
+      for (final ColumnFamilyHandle columnHandle : columnHandles) {
+        final String segmentName = segmentsById.get(BytesValue.wrap(columnHandle.getName()));
+        if (segmentName != null) {
+          builder.put(segmentName, columnHandle);
+        } else {
+          builder.put(DEFAULT_COLUMN, columnHandle);
+        }
+      }
+      columnHandlesByName = builder.build();
+
+      readLatency =
+          metricsSystem
+              .createLabelledTimer(
+                  PantheonMetricCategory.KVSTORE_ROCKSDB,
+                  "read_latency_seconds",
+                  "Latency for read from RocksDB.",
+                  "database")
+              .labels(rocksDbConfiguration.getLabel());
+      removeLatency =
+          metricsSystem
+              .createLabelledTimer(
+                  PantheonMetricCategory.KVSTORE_ROCKSDB,
+                  "remove_latency_seconds",
+                  "Latency of remove requests from RocksDB.",
+                  "database")
+              .labels(rocksDbConfiguration.getLabel());
+      writeLatency =
+          metricsSystem
+              .createLabelledTimer(
+                  PantheonMetricCategory.KVSTORE_ROCKSDB,
+                  "write_latency_seconds",
+                  "Latency for write to RocksDB.",
+                  "database")
+              .labels(rocksDbConfiguration.getLabel());
+      commitLatency =
+          metricsSystem
+              .createLabelledTimer(
+                  PantheonMetricCategory.KVSTORE_ROCKSDB,
+                  "commit_latency_seconds",
+                  "Latency for commits to RocksDB.",
+                  "database")
+              .labels(rocksDbConfiguration.getLabel());
+
+      if (metricsSystem instanceof PrometheusMetricsSystem) {
+        RocksDBStats.registerRocksDBMetrics(stats, (PrometheusMetricsSystem) metricsSystem);
+      }
+
+      metricsSystem.createLongGauge(
+          PantheonMetricCategory.KVSTORE_ROCKSDB,
+          "rocks_db_table_readers_memory_bytes",
+          "Estimated memory used for RocksDB index and filter blocks in bytes",
+          () -> {
+            try {
+              return db.getLongProperty("rocksdb.estimate-table-readers-mem");
```

Comment:
Off topic for another PR - can we get the estimated DB size from another property?  Even if it is a different metric that size on disk?

Reply:
PAN-2853, public and in the backlog.  Probably a good first issue for an external contributor.